docs/decisions/0004-downstream-effect-events.rst

```diff
@@ -0,0 +1,103 @@
+Implementation of Event Driven Architecture for Exam Downstream Effects
+=======================================================================
+
+Status
+------
+
+Draft (circa Sept 2023)
+
+Context
+-------
+For Milestone 6 of the exams IDA project, we are porting any downstream effects of exam submission and review
+from edx-proctoring to the exams IDA (aka edx-exams). Most of these downstream effects will reflect calls
+that edx-proctoring (which is a plugin) makes to various edx-platform services, such as grades, credits, and certificates.
+
+Since edx-exams is independent, we plan to use the event bus to send info to edx-platform services
+without needing a response as one would in a REST setup.
+
+This is being done to circumvent messy circular dependencies, wherein edx-exams and edx-platform go back and
+forth asking each other for information.
+
+
+Decision
+--------
+We are porting these because they are both essential and easy to translate into events:
+
+  * Grades Override - A call to the grades service to override a grade when an exam attempt is rejected.
+
+  * Undo Grades Override - A call to the grades to undo such an override when marking a rejected attempt as verified.
+
+  * Instructor Delete Attempt - A call to the instructor service to delete an exam attempt.
+
+  * Instructor Complete Attempt - A call to the instructor service that marks the exam as completed.
+
+  * Instructor Is Course Staff - A call to the instructor service to check if a user is course staff.
+
+  * Invalidate Certificate - A call to the certificates service to invalidate a leaner’s edx certificate for a course.
+
+  * Set Credit Requirement Status - A call to the credits service to create or modify a learner’s credit requirement status for an exam.
+
+Here’s what we’re NOT porting (and why):
```

Comment:
An ADR isn't always a commitment to completely finish the development it just sets a direction. So things that might happen and in different project or milestone can still be consequences of the decision.

I think the language here should define what should be built instead of stating what 'we' are specifically committing to do.

Instead of "we have decided to implement X" say "x will be implemented"